app/models/asset.rb

```diff
@@ -25,14 +25,21 @@ def with_predicate(predicate)
         super
       end
     end
+
+    def predicate_matching?(predicate, value)
+      if loaded?
+        with_predicate(predicate).any? { |fact| fact.object == value }
+      else
+        exists?(predicate: 'a', object: 'Plate')
+      end
+    end
```

Comment:
Allows us to use the eager-loaded facts when checking his if they are available. Avoids additional trips to the database.